ports/stm32/eth.c

```diff
@@ -26,12 +26,14 @@
 
 #include <string.h>
 #include "py/mphal.h"
+#include "py/runtime.h"
```

Comment:
Why is this header needed?

Reply:
For raising an error about a wrong phy_type:
`mp_raise_ValueError(MP_ERROR_TEXT("Invalid phy_type"));`

Reply:
I see.

It would be better to separate the layers here: low-level hardware init vs higher-level Python interface.

I suggest changing `eth_init()` to return `int`, with `0` being success and a negative number being an error code.  Then callers of this function must test this error code and raise as appropriate.

As part of this, note that `ARDUINO_PORTENTA_H7`'s board init function calls `eth_init()`, and that should definitely not raise an exception (because that code runs outside the main MicroPython framework).  So that use will need to be updated to pass the correct PHY.

Reply:
Moved the raise() call to network_lan.c and added the proper parameters to board_init.c of the Portenta H7. Thank you for the hint about the Portenta H7. I did not expect that.